streams/src/main/java/org/apache/kafka/streams/processor/internals/StateDirectory.java

```diff
@@ -107,7 +106,7 @@ public StateDirectoryProcessFile() {
     private final boolean hasPersistentStores;
     private final boolean hasNamedTopologies;
 
-    private final HashMap<TaskId, Thread> lockedTasksToOwner = new HashMap<>();
+    private final ConcurrentMap<TaskId, Thread> lockedTasksToOwner = new ConcurrentHashMap<>();
```

Comment:
side fix : with different thread accessing this map I think it makes sense to change it to a `ConcurrentMap`